app/src/main/java/com/github/scribeWizTeam/scribewiz/fragments/RecFragment.kt

```diff
@@ -201,10 +205,19 @@ class RecFragment(
     }
 
     private fun switchRecordState(
+        context: Context,
         counterText: MutableState<String>,
         recordButtonText: MutableState<String>
     ) {
+        // Check for audio recording permissions
+        if (ContextCompat.checkSelfPermission(context, Manifest.permission.RECORD_AUDIO) != PackageManager.PERMISSION_GRANTED) {
+            // No permissions, send a toast
+            Toast.makeText(context, "Microphone permission required", Toast.LENGTH_LONG).show()
```

Comment:
Good idea, it may be good to have toast feedback for action like submitting to a challenge too !